query-engine/js-connectors/src/queryable.rs

```diff
@@ -191,14 +152,136 @@ impl JsQueryable {
     }
 }
 
-impl TransactionCapable for JsQueryable {}
+/// A JsQueryable adapts a Proxy to implement quaint's Queryable interface. It has the
+/// responsibility of transforming inputs and outputs of `query` and `execute` methods from quaint
+/// types to types that can be translated into javascript and viceversa. This is to let the rest of
+/// the query engine work as if it was using quaint itself. The aforementioned transformations are:
+///
+/// Transforming a `quaint::ast::Query` into SQL by visiting it for the specific flavour of SQL
+/// expected by the client connector. (eg. using the mysql visitor for the Planetscale client
+/// connector)
+///
+/// Transforming a `JSResultSet` (what client connectors implemented in javascript provide)
+/// into a `quaint::connector::result_set::ResultSet`. A quaint `ResultSet` is basically a vector
+/// of `quaint::Value` but said type is a tagged enum, with non-unit variants that cannot be converted to javascript as is.
+///
+#[derive(Clone)]
+pub struct JsQueryable {
+    inner: JsBaseQueryable,
+    driver_proxy: DriverProxy,
+}
+
+impl std::fmt::Display for JsQueryable {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "JSQueryable(driver)")
+    }
+}
+
+impl std::fmt::Debug for JsQueryable {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "JSQueryable(driver)")
+    }
+}
+
+#[async_trait]
+impl QuaintQueryable for JsQueryable {
+    /// Execute the given query.
```

Comment:
Nitpick, but we could remove the doc comments in this block: they can get out of sync with the trait declaration docs. The language server will show the docs from the trait anyway.